connect/client/models/iterators.py

```diff
@@ -124,6 +124,58 @@ async def _execute_request(self):
         return results, content_range
 
 
+class AbstractSliceIterator(AbstractBaseIterator):
```

Comment:
IMO this logic can be added to the AbstractIteration instead of having a new iterator class for slices, am I wrong ?

Reply:
AbstractIteration caches results, tests showed, that it leads to different error. Moreover, AbstractIteration don't have   `def __iter__(self):`